flik/linesearch/interpolations.py

```diff
@@ -0,0 +1,109 @@
+    interval_end: float
+        The ending point of the interval containing desirable step lengths
+    alpha_i_: float
+        The (i-1)-th step length estimate
+    alpha_i: float
+        The i-th step length estimate
+
+    Raises
+    ------
+        TypeError
+            according to the function check_input
+        ValueError
+            according to the function check_input
+            if interval_start and interval_end are not such that
+                0 < interval_start < interval_end <= 1
+            if alpha_1 and alpha_2 are not such that
+                interval_start < alpha_i_ < alpha_i < interval_end
+
+    Returns
+    -------
+    alpha: float
+        The minimizer :math:`\alpha` of :math:`f(x + \alpha p)`, where alpha is in the provided interval
+        [interval_start, interval_end]
+    """
+
+    # Check the input arguments
+    # TODO: Check the parameters according to the function check_input
+    if isinstance(var, float):
+        var = np.array(var)
+    elif not (isinstance(var, np.ndarray) and var.ndim == 1):
+        raise TypeError("Variable vector should be a float or a 1-D numpy.ndarray")
+    if not callable(func):
+        raise TypeError("func must be callable")
+    if not callable(grad):
+        raise TypeError("grad must be callable")
+    if isinstance(direction, float):
+        direction = np.array(direction)
+    elif not (isinstance(direction, np.ndarray) and direction.ndim == 1):
+        raise TypeError("The direction vector should be provided as a float or"
+                        " a numpy.ndarray")
+    if var.shape != direction.shape:
+        raise ValueError("The shape of vec and direction should be the same")
+
+    #   Check for correct alphas
+    if not isinstance(alpha_i_, float):
+        raise TypeError("Alpha_i_ should be provided as a float")
+    if not isinstance(alpha_i, float):
+        raise TypeError("Alpha_i should be provided as a float")
+
+    #   Check for 0 < interval_start < interval_end <= 1
+    if not (0.0 < interval_start < interval_end <= 1.0):
+        raise ValueError("interval_start and interval_end should be according to 0 < interval_start < interval_end <= 1")
+
+    #   Check for interval_start < alpha_1 < alpha_2 < interval_end
+    if not (interval_start < alpha_i_ < alpha_i < interval_end):
+        raise ValueError("alpha_i_ and alpha_i should be according interval_start < alpha_i_ < alpha_i < interval_end")
+
+    # Define the helper functions phi and its derivative
+    def phi(alpha):
+        return func(var + alpha * direction)
+
+    def phi_(alpha):
+        """The derivative of the previously defined function phi."""
+        return grad(var + alpha * direction).dot(direction)
+
+
+    print(var)
+    print(direction)
+    print(alpha_i_, phi(alpha_i_), phi_(alpha_i_))
+    print(alpha_i, phi(alpha_i), phi_(alpha_i))
+    print(interval_start, phi(interval_start))
+    print(interval_end, phi(interval_end))
```

Comment:
I think we can delete all of the print statements (we'll do a proper logging later on I think)